app/components/validated-input/component.ts

```diff
@@ -0,0 +1,76 @@
+// This component is derived from ember-cp-validations.
+// See https://github.com/offirgolan/ember-cp-validations for more information
+import Component from '@ember/component';
+import { computed, defineProperty } from '@ember/object';
+import { alias, and, not, notEmpty, oneWay } from '@ember/object/computed';
+import { isEmpty } from '@ember/utils';
+import DS from 'ember-data';
+
+enum InputType {
+    Date = 'date',
+    Password = 'password',
+    Text = 'text',
+    TextArea = 'textarea',
+    ReCaptcha = 'reCaptcha',
+}
+
+export default class ValidatedInput extends Component.extend({
+    classNames: ['validated-input'],
+    classNameBindings: ['showErrorClass:has-error', 'isValid:has-success'],
+
+    disabled: false,
```

Comment:
It'd be nice to move these initializations to the class body, where they're defined/typed. The lazy way I've done it before is:
```ts
disabled: boolean = this.disabled || false;
```
but that's brittle for falsy values. I've seen some people use lodash:
```ts
disabled: boolean = _.defaultTo(this.disabled, false);
```
but that's yet another dependency. Could make our own `defaultTo` util, I guess, but that's still an extra import everywhere... Not sure, but that might be more appealing to me than these duplicate lists of properties.